src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunTableManager.cs

```diff
@@ -53,12 +55,24 @@ protected virtual void Graph_NewMarkedNode(DependencyNodeCore<NodeFactory> obj)
 
             if (methodNode != null)
             {
-                _methodsGenerated.Add(methodNode.Method);
+                lock (_methodsGenerated)
+                {
+                    Debug.Assert(_methodsGenerated != null);
```

Comment:
>Debug.Assert(_methodsGenerated != null); [](start = 20, length = 40)

If you want to have this assert, you should put it outside of the lock.

Reply:
Then it's possible for us to observe that _methodsGenerated is not null, have it become null, and then try to add to it, isn't it?

Reply:
If `_methodsGenerated` is null, you'll get an AV on the lock anyways. This makes the assert useless.

Reply:
Ah right. Changed.